data_model/src/events/execute_trigger.rs

```diff
@@ -53,21 +53,60 @@ pub mod model {
     )]
     pub struct ExecuteTriggerEventFilter {
         /// Id of trigger catch executions of
-        pub(super) trigger_id: TriggerId,
+        pub(super) trigger_id: Option<TriggerId>,
         /// Authority of user who owns trigger
-        pub(super) authority: AccountId,
+        pub(super) authority: Option<AccountId>,
+    }
+}
+
+impl ExecuteTriggerEventFilter {
+    /// Creates a new [`ExecuteTriggerEventFilter`] accepting all [`ExecuteTriggerEvent`]s
+    #[must_use]
+    #[inline]
+    pub const fn new() -> Self {
```

Comment:
```suggestion
    pub const fn any() -> Self {
```

wouldn't this be more appropriate? In other places as well

Reply:
yes, this is what I would prefer. I think this API is more easy to understand

Reply:
I actually strongly prefer the `::new()`, even more so with the change to use `for_*` (which I very much like because it brings more consistency to all the filter types):

```rust
AccountEventFilter::new()
    .for_events(AccountEventSet::Created)
    .for_account("alice@wonderland".parse()?)
```

While I do think that clarifying via the method name that the default state of the filter is "accept all events", this is _the_ natural state for it to be in (and you can look it up in method docs).

There are several reasons I dislike `any` though:
- `new` reads better as a sentence: "new event filter for events XYZ and account W". With `any`, it's difficult to construct such a sentence.
- more complex mental model needed when constructing event filters: "I need to make a filter that accepts all events and restrict it to only accept events from this account" vs "I need to make a filter only accepting events from this account".
- `new` is consistent with builder pattern: you first create a new object and then modify it with some method calls. The constructor being called `any` instead of `new` breaks the pattern, with marginal benefits